src/services/PostEventProcessor.ts

```diff
@@ -41,164 +39,157 @@ export class PostEventProcessor {
-				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Missing info in sqs event");
+	
+			const { user_id, govuk_signin_journey_id } = user;
+			if (!user_id) {
+					const errorMessage = "Missing or invalid value for userDetails.user_id in event payload";
+					this.logger.error({ message: errorMessage }, { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS });
+					throw new AppError(HttpCodesEnum.SERVER_ERROR, errorMessage);
 			}
-			const userDetails = eventDetails.user;
-			this.logger.appendKeys({ govuk_signin_journey_id: userDetails.govuk_signin_journey_id });
-
-			if (!this.checkIfValidString([userDetails.user_id])) {
-				this.logger.error({ message: "Missing or invalid value for userDetails.user_id in event payload" }, { messageCode: MessageCodes.MISSING_MANDATORY_FIELDS });
-				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Missing info in sqs event");
-			}
-
-			const userId = userDetails.user_id;
-			// Do not process the event if the event is already processed or flagged for deletion
-			const isFlaggedForDeletionOrEventAlreadyProcessed = await this.iprService.isFlaggedForDeletionOrEventAlreadyProcessed(userId, eventName);
+	
+			this.logger.appendKeys({ event_id, govuk_signin_journey_id });
+			this.logger.info({ message: `Received SQS event with eventName ${event_name}` });
+	
+			const isFlaggedForDeletionOrEventAlreadyProcessed = await this.iprService.isFlaggedForDeletionOrEventAlreadyProcessed(user_id, event_name);
+	
 			if (isFlaggedForDeletionOrEventAlreadyProcessed) {
-				this.logger.info( { message: "Record flagged for deletion or event already processed, skipping update" });
-				return "Record flagged for deletion or event already processed, skipping update";
+					this.logger.info({ message: "Record flagged for deletion or event already processed, skipping update" });
+					return "Record flagged for deletion or event already processed, skipping update";
 			}
-
-			let updateExpression, expressionAttributeValues: { [key: string]: any }, expiresOn;
-
-			//Set default TTL to 12hrs to expire any records not meant for F2F
-			expiresOn = absoluteTimeNow() + this.environmentVariables.initialSessionReturnRecordTtlSecs();
-
-			if (eventName === Constants.F2F_YOTI_START) {
-				//Reset TTL to 11days for F2F journey
-				expiresOn = absoluteTimeNow() + this.environmentVariables.sessionReturnRecordTtlSecs();
+	
+			let updateExpression, expressionAttributeValues: { [key: string]: any };
+	
+			const expiresOn = absoluteTimeNow() + (event_name === Constants.F2F_YOTI_START ? this.environmentVariables.sessionReturnRecordTtlSecs() : this.environmentVariables.initialSessionReturnRecordTtlSecs());
+	
+			const returnRecord = new SessionReturnRecord(eventDetails, expiresOn);
+
+			const { AUTH_REQUESTED, YOTI_START, VC_CONSUMED, DOCUMENT_UPLOADED, DELETE_ACCOUNT } = TXMA_EVENT_DETAILS;
+	
+			switch (event_name) {
```

Comment:
By its very nature this processor takes lots of different events and does different things with them - and apologies is this is a tangent - what do you think about splitting it out or using step functions or something so that we start to separate some concerns?  Would it create unnecessary overhead? Or be easier to reason about?

Reply:
Maybe another day..